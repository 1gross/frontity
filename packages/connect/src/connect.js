import { createContext, useContext, createElement } from "react";
import { view } from "@risingstack/react-easy-state";

/**
 * A React context that stores the store of Frontity Connect.
 */
const context = createContext();

/**
 * The Provider of the store context, to be used in the root of the React
 * application.
 */
export const Provider = context.Provider;

/**
 * Connects a React component with the Frontity connect store.
 *
 * @param Component The React component to be connected
 * @returns The same Component, but with the store included in the props.
 */
export const connect = (Component) => {
  const isStatelessComp = !(
    Component.prototype && Component.prototype.isReactComponent
  );

  if (isStatelessComp) {
    return view((props) => {
      const store = useContext(context);
      return Component({ ...props, ...store });
    });
  } else {
    class ConnectedComponent extends Component {
      static contextType = context;

      render() {
        return createElement(Component, {
          ...this.props,
          ...this.context,
        });
      }
    }

    return view(ConnectedComponent);
  }
};
