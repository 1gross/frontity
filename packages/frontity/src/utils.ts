import { readdirSync } from "fs";

// This function checks if the path passed is an empty directory
// or an empty GitHub repository.
export const isEmptyDir = (path: string = process.cwd()): boolean => {
  try {
    // Get the content of the path.
    const dirContent = readdirSync(path);

    // Return true if the path is empty.
    if (!dirContent.length) return true;

    // Define a list of files/dirs allowed in an empty GitHub repo.
    const allowedContent = ["readme.md", "license", ".gitignore", ".git"];

    // Return true if all the content in path is allowed.
    if (
      !dirContent.filter(
        content => !allowedContent.includes(content.toLowerCase())
      ).length
    )
      return true;
  } catch (error) {
    // Throw if the path does not exist.
    if (error.code === "ENOENT")
      throw new Error("The path provided to `isEmptyDir` does not exist.");
  }

  // Return false if the path is not empty or does not contain an
  // empty GitHub repo.
  return false;
};

// This functions checks if the path passed is a Git repository.
export const isGitRepository = (path: string = process.cwd()): boolean => {
  try {
    // Gets the content of the path.
    const dirContent = readdirSync(path);

    // Returns true if the dir contains a `.git` folder.
    if (dirContent.includes(".git")) return true;
  } catch (error) {
    if (error.code === "ENOENT")
      throw new Error("The path provided to `isGitRepository` does not exist.");
  }

  // Returns false if the dir does not contain a `.git` folder.
  return false;
};
