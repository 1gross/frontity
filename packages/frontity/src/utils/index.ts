export { default as checkMandatoryParams } from "./check-mandatory-params";
export { default as isEmptyDir } from "./is-empty-dir";
export { default as isGitRepository } from "./is-git-repository";
export { default as normalizeOptions } from "./normalize-options";
