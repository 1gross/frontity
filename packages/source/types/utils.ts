/**
 *
 *
 * @param M First type.
 * @param N - Second type.
 * @return Merged types.
 */
export type Merge<M, N> = Omit<M, Extract<keyof M, keyof N>> & N;
