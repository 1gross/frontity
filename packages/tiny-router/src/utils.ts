export const parsePath = (
  pathOrLink: string
): { path: string; page: number } => {
  let fullPath: string;

  try {
    // Try if it's a URL.
    fullPath = new URL(pathOrLink).pathname;
  } catch (e) {
    // It's already a path.
    fullPath = pathOrLink;
  }

  const path = (/^(.*)\/page\//.exec(fullPath) || [null, fullPath])[1];
  const page = parseInt(
    (/\/page\/(\d+)\/?$/.exec(fullPath) || [null, "1"])[1],
    10
  );

  return { path, page };
};
