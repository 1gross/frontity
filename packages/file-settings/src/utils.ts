import fs from "fs";
import path from "path";
import { AllSettings } from "./types";

// This function imports the available settings, either custom or default.
export const importSettingsFromFile = async (): Promise<AllSettings> => {
  // Path to the custom settings file.
  const customSettingsPath: string = process.cwd();
  // Path to the default settings file.
  const defaultSettingsPath: string = path.resolve(
    process.cwd(),
    "node_modules/@frontity/file-settings"
  );

  // Possible settings files and its priorities:
  // 1. Custom settings in TypeScript.
  // 2. Custom settings in JavaScript.
  // 3. Default settings in TypeScript.
  // 4. Default settings in JavaScript.
  const settingsPaths: string[] = [
    path.resolve(customSettingsPath, "frontity.settings.ts"),
    path.resolve(customSettingsPath, "frontity.settings.js"),
    path.resolve(defaultSettingsPath, "frontity.settings.ts"),
    path.resolve(defaultSettingsPath, "frontity.settings.js")
  ];

  // Available settings file.
  let settingsFile: string;

  // Loop over the possible settings files until one is available.
  for (let settingsPath of settingsPaths) {
    const fileExists: boolean = fs.existsSync(settingsPath);

    if (fileExists) {
      settingsFile = settingsPath;
      break;
    }
  }

  // Return the settings imported from the available file.
  return (await import(settingsFile)).default;
};
