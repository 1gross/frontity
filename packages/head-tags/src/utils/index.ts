import { URL } from "frontity";
import { HeadTags, State } from "../../types";

type GetUrlPathname = (url: URL, apiUrl: URL, isWpCom: boolean) => string;

type UseFrontityLinks = (args: {
  state: State;
  headTags: HeadTags;
}) => HeadTags;

// Attributes that could contain links.
const possibleLink = ["href", "content"];

// Test if a path is not from a blog link.
const isInvalid = /^\/wp-(json|admin|content)/;

export const getUrlPathname: GetUrlPathname = (url, apiUrl, isWpCom) => {
  // Get API subdirectory.
  const apiSubdir = isWpCom
    ? ""
    : apiUrl.pathname.replace(/\/wp-json\/?$/, "/");

  // Remove the API subdirectory.
  let { pathname } = url;
  if (apiSubdir && pathname.startsWith(apiSubdir))
    pathname = pathname.replace(apiSubdir, "/");

  // Throw an error (to be captured by `useFrontityLinks`).
  if (isInvalid.test(pathname)) throw new Error("Invalid pathname");

  // Return the final path.
  return pathname === "/" ? "" : pathname;
};

export const useFrontityLinks: UseFrontityLinks = ({ state, headTags }) => {
  // The site URL.
  const frontityUrl = state.frontity.url.replace(/\/?$/, "");

  // The API URL.
  const apiUrl = new URL(state.source.api);
  const { isWpCom } = state.source;

  // For each head tag...
  return headTags.map(({ tag, attributes, content }) => {
    // Compute new attributes changing URLs.
    const newAttributes = Object.entries(attributes).reduce(
      (result, [key, value]) => {
        if (possibleLink.includes(key)) {
          try {
            // Transform value into a URL (if possible).
            const url = new URL(value);

            // Get the path of this URL (THROWS AN ERROR IF IS NOT A VALID ONE).
            const pathname = getUrlPathname(url, apiUrl, isWpCom);

            // Set Frontity URL.
            const { search, hash } = url;
            result[key] = `${frontityUrl}${pathname}${search}${hash}`;
          } catch (e) {
            // Don't change the value.
            result[key] = value;
          }
        }
        return result;
      },
      {}
    );

    // Replace attributes.
    return { tag, attributes: newAttributes, content };
  });
};
