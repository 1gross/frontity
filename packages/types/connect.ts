import Package from "./package";
import { ResolveState, Omit } from "./utils";

// Resolves the actions for things like Action and Derived.
export type ResolveActions<Actions extends Package["state"]> = {
  [P in keyof Actions]: Actions[P] extends (
    ...a: any
  ) => (arg: infer Arg) => void // Turns "state => args => {}" into "args => {}"
    ? (arg: Arg) => void // Turns "state => {}" into "() => {}"
    : Actions[P] extends (state: Package["state"]) => any
    ? () => void
    : ResolveActions<Actions[P]>
};

export type Connect<S extends Package, Props extends object = {}> = Omit<
  S,
  "state" | "actions" | "name"
> & {
  state: ResolveState<S["state"]>;
  actions: ResolveActions<S["actions"]>;
} & Props;

export default Connect;
