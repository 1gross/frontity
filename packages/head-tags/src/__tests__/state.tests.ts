/* eslint-disable @typescript-eslint/camelcase */
import clone from "clone-deep";
import { createStore, InitializedStore } from "@frontity/connect";
import wpSource from "@frontity/wp-source/src";
import HeadTagsPackage, { State, HeadTags } from "../../types";
import headTagsPackage from "..";
import { mockPostEntity } from "./mocks/utils";

let store: InitializedStore<HeadTagsPackage>;
beforeEach(() => {
  // Create store.
  const config: HeadTagsPackage = clone(headTagsPackage());

  // Mock wp-source state.
  config.state.source = clone(wpSource()).state.source;
  config.state.source.api = "https://test.frontity.io/wp-json";
  // Mock router state.
  config.state.router = { link: "/" };
  // Mock site url.
  config.state.frontity = { url: "https://mars.frontity.org" };
  // Initialize store.
  store = createStore(config);
});

describe("Current Head Tags - Post Entity", () => {
  const setUpState = (state: State, headTags?: HeadTags) => {
    // Populate source state.
    const { post, data } = mockPostEntity(headTags);
    state.source.post = post;
    state.source.data = data;

    // Populate router state.
    state.router.link = "/post-1/";
  };

  test("returns an empty array if `head_tags` is undefined", () => {
    const headTags = undefined;
    // Populate all state.
    setUpState(store.state, headTags);
    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("returns elements without attributes (title)", () => {
    const headTags: HeadTags = [{ tag: "title", content: "Post 1 - Frontity" }];
    // Populate all state.
    setUpState(store.state, headTags);
    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("returns elements without content (meta tags)", () => {
    const headTags: HeadTags = [
      {
        tag: "meta",
        attributes: {
          name: "robots",
          content:
            "max-snippet:-1, max-image-preview:large, max-video-preview:-1"
        }
      }
    ];
    // Populate all state.
    setUpState(store.state, headTags);
    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("transforms links that point to WordPress pages", () => {
    const headTags: HeadTags = [
      {
        tag: "link",
        attributes: {
          rel: "canonical",
          href: "https://test.frontity.io/post-1/"
        }
      },
      {
        tag: "meta",
        attributes: {
          property: "og:url",
          content: "https://test.frontity.io/post-1/"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "shortlink",
          href: "https://test.frontity.io/?p=1"
        }
      }
    ];
    // Populate all state.
    setUpState(store.state, headTags);
    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("doesn't change links that don't point to WordPress pages", () => {
    const headTags: HeadTags = [
      {
        tag: "meta",
        attributes: {
          property: "og:image",
          content: "https://test.frontity.io/wp-content/uploads/2019/12/img.jpg"
        }
      },
      {
        tag: "meta",
        attributes: {
          name: "twitter:image",
          content: "https://test.frontity.io/wp-content/uploads/2019/12/img.jpg"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "dns-prefetch",
          href: "//s.w.org"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "alternate",
          type: "application/rss+xml",
          title: "frontity » Feed",
          href: "https://test.frontity.io/feed/"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "alternate",
          type: "application/rss+xml",
          title: "frontity » Comments Feed",
          href: "https://test.frontity.io/comments/feed/"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "alternate",
          type: "application/rss+xml",
          title: "frontity » Post 1 Comments Feed",
          href: "https://test.frontity.io/post-1/feed/"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "https://api.w.org/",
          href: "https://test.frontity.io/wp-json/"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "EditURI",
          type: "application/rsd+xml",
          title: "RSD",
          href: "https://test.frontity.io/xmlrpc.php?rsd"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "wlwmanifest",
          type: "application/wlwmanifest+xml",
          href: "https://test.frontity.io/wp-includes/wlwmanifest.xml"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "alternate",
          type: "application/json+oembed",
          href:
            "https://test.frontity.io/wp-json/oembed/1.0/embed?url=http%3A%2F%2Ftest.frontity.io%2Fpost-1%2F"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "alternate",
          type: "text/xml+oembed",
          href:
            "https://test.frontity.io/wp-json/oembed/1.0/embed?url=http%3A%2F%2Ftest.frontity.io%2Fpost-1%2F&format=xml"
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "pingback",
          href: "https://test.frontity.io/xmlrpc.php"
        }
      }
    ];
    // Populate all state.
    setUpState(store.state, headTags);
    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("transform links inside ld+json data", () => {
    const headTags: HeadTags = [
      {
        tag: "script",
        attributes: {
          type: "application/ld+json",
          class: "yoast-schema-graph yoast-schema-graph--main"
        },
        content: JSON.stringify({
          "@context": "https://schema.org",
          "@graph": [
            {
              // All these links should change...
              "@type": "WebSite",
              "@id": "https://test.frontity.io/#website",
              url: "https://test.frontity.io/",
              potentialAction: {
                "@type": "SearchAction",
                target: "https://test.frontity.io/?s={search_term_string}"
              }
            },
            {
              "@type": "ImageObject",
              "@id": "https://test.frontity.io/post-1/#primaryimage",
              // ...except this one.
              url: "https://test.frontity.io/wp-content/uploads/2019/12/img.jpg"
            }
          ]
        })
      }
    ];
    // Populate all state.
    setUpState(store.state, headTags);
    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("transforms links appropiately when WP is in a subdirectory", () => {
    const headTags: HeadTags = [
      {
        tag: "link",
        attributes: {
          rel: "canonical",
          href: "https://test.frontity.io/subdir/post-1/" // should change
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "https://api.w.org/",
          href: "https://test.frontity.io/subdir/wp-json/" // should not change
        }
      },
      {
        tag: "script",
        attributes: {
          type: "application/ld+json"
        },
        content: JSON.stringify({
          "@context": "https://schema.org",
          "@graph": [
            {
              // All these links should change
              "@type": "WebSite",
              "@id": "https://test.frontity.io/subdir/#website",
              url: "https://test.frontity.io/subdir/"
            }
          ]
        })
      }
    ];
    // Populate all state.
    setUpState(store.state, headTags);

    // Change API prop
    store.state.source.api = "https://test.frontity.io/subdir/wp-json/";

    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });

  test("transforms links appropiately when Frontity is in a subdirectory", () => {
    const headTags: HeadTags = [
      {
        tag: "link",
        attributes: {
          rel: "canonical",
          href: "https://test.frontity.io/subdir/post-1/" // should change
        }
      },
      {
        tag: "link",
        attributes: {
          rel: "https://api.w.org/",
          href: "https://test.frontity.io/subdir/wp-json/" // should not change
        }
      },
      {
        tag: "script",
        attributes: {
          type: "application/ld+json"
        },
        content: JSON.stringify({
          "@context": "https://schema.org",
          "@graph": [
            {
              // All these links should change
              "@type": "WebSite",
              "@id": "https://test.frontity.io/subdir/#website",
              url: "https://test.frontity.io/subdir/"
            }
          ]
        })
      }
    ];
    // Populate all state.
    setUpState(store.state, headTags);

    // Change API and Frontity URL prop
    store.state.frontity.url = "https://frontity.org/mars/";
    store.state.source.api = "https://test.frontity.io/subdir/wp-json/";

    // Test current head tags.
    expect(store.state.headTags.current).toMatchSnapshot();
  });
});
