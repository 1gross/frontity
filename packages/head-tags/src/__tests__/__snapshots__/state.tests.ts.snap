// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state.headTags.current (author) works with auhors 1`] = `
Array [
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/author/author-1/",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/author/author-1/page/2/",
      "rel": "next",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://mars.frontity.org#website\\",\\"url\\":\\"https://mars.frontity.org\\"}]}",
    "tag": "script",
  },
]
`;

exports[`state.headTags.current (post entity) doesn't change links that don't point to WordPress pages 1`] = `
Array [
  Object {
    "attributes": Object {
      "content": "https://test.frontity.io/wp-content/uploads/2019/12/img.jpg",
      "property": "og:image",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "https://test.frontity.io/wp-content/uploads/2019/12/img.jpg",
      "name": "twitter:image",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "href": "//s.w.org",
      "rel": "dns-prefetch",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/feed/",
      "rel": "alternate",
      "title": "frontity » Feed",
      "type": "application/rss+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/comments/feed/",
      "rel": "alternate",
      "title": "frontity » Comments Feed",
      "type": "application/rss+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/post-1/feed/",
      "rel": "alternate",
      "title": "frontity » Post 1 Comments Feed",
      "type": "application/rss+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/xmlrpc.php?rsd",
      "rel": "EditURI",
      "title": "RSD",
      "type": "application/rsd+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-includes/wlwmanifest.xml",
      "rel": "wlwmanifest",
      "type": "application/wlwmanifest+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/oembed/1.0/embed?url=http%3A%2F%2Ftest.frontity.io%2Fpost-1%2F",
      "rel": "alternate",
      "type": "application/json+oembed",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/oembed/1.0/embed?url=http%3A%2F%2Ftest.frontity.io%2Fpost-1%2F&format=xml",
      "rel": "alternate",
      "type": "text/xml+oembed",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/xmlrpc.php",
      "rel": "pingback",
    },
    "tag": "link",
  },
]
`;

exports[`state.headTags.current (post entity) returns an empty array if \`head_tags\` is undefined 1`] = `Array []`;

exports[`state.headTags.current (post entity) returns elements without attributes (title) 1`] = `
Array [
  Object {
    "content": "Post 1 - Frontity",
    "tag": "title",
  },
]
`;

exports[`state.headTags.current (post entity) returns elements without content (meta tags) 1`] = `
Array [
  Object {
    "attributes": Object {
      "content": "max-snippet:-1, max-image-preview:large, max-video-preview:-1",
      "name": "robots",
    },
    "tag": "meta",
  },
]
`;

exports[`state.headTags.current (post entity) transform links inside ld+json data 1`] = `
Array [
  Object {
    "attributes": Object {
      "class": "yoast-schema-graph yoast-schema-graph--main",
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://mars.frontity.org#website\\",\\"url\\":\\"https://mars.frontity.org\\",\\"potentialAction\\":{\\"@type\\":\\"SearchAction\\",\\"target\\":\\"https://mars.frontity.org?s={search_term_string}\\"}},{\\"@type\\":\\"ImageObject\\",\\"@id\\":\\"https://mars.frontity.org/post-1/#primaryimage\\",\\"url\\":\\"https://test.frontity.io/wp-content/uploads/2019/12/img.jpg\\"}]}",
    "tag": "script",
  },
]
`;

exports[`state.headTags.current (post entity) transforms links appropiately when Frontity is in a subdirectory 1`] = `
Array [
  Object {
    "attributes": Object {
      "href": "https://frontity.org/mars/post-1/",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/subdir/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://frontity.org/mars#website\\",\\"url\\":\\"https://frontity.org/mars\\"}]}",
    "tag": "script",
  },
]
`;

exports[`state.headTags.current (post entity) transforms links appropiately when WP is in a subdirectory 1`] = `
Array [
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/post-1/",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/subdir/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://mars.frontity.org#website\\",\\"url\\":\\"https://mars.frontity.org\\"}]}",
    "tag": "script",
  },
]
`;

exports[`state.headTags.current (post entity) transforms links that point to WordPress pages 1`] = `
Array [
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/post-1/",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "content": "https://mars.frontity.org/post-1/",
      "property": "og:url",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org?p=1",
      "rel": "shortlink",
    },
    "tag": "link",
  },
]
`;

exports[`state.headTags.current (post type) works with post types 1`] = `
Array [
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/page/2/",
      "rel": "next",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://mars.frontity.org#website\\",\\"url\\":\\"https://mars.frontity.org\\"}]}",
    "tag": "script",
  },
]
`;

exports[`state.headTags.current (taxonomy) works with taxonomies 1`] = `
Array [
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/category/cat-1/",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/category/cat-1/page/2/",
      "rel": "next",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://mars.frontity.org#website\\",\\"url\\":\\"https://mars.frontity.org\\"}]}",
    "tag": "script",
  },
]
`;
